client/my-sites/stats/activity-log-search/activity-log-search-tokens.jsx

```diff
@@ -3,18 +3,36 @@
 /**
  * External dependencies
  */
-
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
 import PropTypes from 'prop-types';
 import React, { Component } from 'react';
-import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { filterStateToTokens, tokensToFilterState } from 'state/activity-log/utils';
+import TokenField from 'components/token-field';
+import { setFilter } from 'state/activity-log/actions';
 
 class ActivityLogSearchTokens extends Component {
 	static propTypes = {
 		filter: PropTypes.object.isRequired,
 	};
+
+	onTokensChange = tokens => {
+		this.props.setFilter( this.props.siteId, tokensToFilterState( tokens ) );
+	};
+
 	render() {
-		return <p>tokens</p>;
+		const tokens = filterStateToTokens( this.props.filter );
+		return <TokenField value={ tokens } onChange={ this.onTokensChange } />;
 	}
 }
 
-export default localize( ActivityLogSearchTokens );
+export default connect(
+	() => {
+		return {};
+	},
```

Comment:
`null` ??

Reply:
😆 - yeah, i have not built a component that only needs connected actions, so i wasn't sure what to put